akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupedWithinSpec.scala

```diff
@@ -157,4 +172,87 @@ class FlowGroupedWithinSpec extends StreamSpec with ScriptedTest {
+      val downstream = TestSubscriber.probe[immutable.Seq[Int]]()
+      Source(List(1, 2, 3, 101, 4, 5, 6))
+        .groupedWeightedWithin(100, 100.millis)(_.toLong)
+        .to(Sink.fromSubscriber(downstream))
+        .run()
+
+      downstream.request(1)
+      downstream.expectNext((1 to 3).toVector)
+      downstream.request(1)
+      downstream.expectNext(Vector(101))
+      downstream.request(1)
+      downstream.expectNext((4 to 6).toVector)
+      downstream.expectComplete()
+      downstream.expectNoMsg(100.millis)
+    }
+
+    "not drop a pending last element on upstream finish" taggedAs TimingTest in {
+      val וupstream = TestPublisher.probe[Long]()
+      val downstream = TestSubscriber.probe[immutable.Seq[Long]]()
+      Source
+        .fromPublisher(וupstream)
+        .groupedWeightedWithin(5, 50.millis)(identity)
+        .to(Sink.fromSubscriber(downstream))
+        .run()
+
+      downstream.ensureSubscription()
+      downstream.expectNoMsg(100.millis)
+      וupstream.sendNext(1)
+      וupstream.sendNext(2)
+      וupstream.sendNext(3)
+      וupstream.sendComplete()
+      downstream.request(1)
+      downstream.expectNext(Vector(1, 2): immutable.Seq[Long])
+      downstream.expectNoMsg(100.millis)
+      downstream.request(1)
+      downstream.expectNext(Vector(3): immutable.Seq[Long])
+      downstream.expectComplete()
+    }
+
+    "append zero weighted elements to a full group before timeout received, if downstream hasn't pulled yet" taggedAs TimingTest in {
+      val וupstream = TestPublisher.probe[String]()
+      val downstream = TestSubscriber.probe[immutable.Seq[String]]()
+      Source
+        .fromPublisher(וupstream)
+        .groupedWeightedWithin(5, 50.millis)(_.length.toLong)
+        .to(Sink.fromSubscriber(downstream))
+        .run()
+
+      downstream.ensureSubscription()
+      וupstream.sendNext("333")
+      וupstream.sendNext("22")
+      וupstream.sendNext("")
+      וupstream.sendNext("")
+      וupstream.sendNext("")
+      downstream.request(1)
+      downstream.expectNext(Vector("333", "22", "", "", ""): immutable.Seq[String])
+      וupstream.sendNext("")
+      וupstream.sendNext("")
+      וupstream.sendComplete()
+      downstream.request(1)
+      downstream.expectNext(Vector("", ""): immutable.Seq[String])
+      downstream.expectComplete()
+    }
+
+    "not emit an empty group if first element is heavier than maxWeight" taggedAs TimingTest in {
+      val וupstream = TestPublisher.probe[Long]()
+      val downstream = TestSubscriber.probe[immutable.Seq[Long]]()
+      Source
+        .fromPublisher(וupstream)
+        .groupedWeightedWithin(10, 50.millis)(identity)
+        .to(Sink.fromSubscriber(downstream))
+        .run()
+
+      downstream.ensureSubscription()
+      downstream.request(1)
+      וupstream.sendNext(11)
+      downstream.expectNext(Vector(11): immutable.Seq[Long])
+      וupstream.sendComplete()
+      downstream.expectComplete()
+    }
+  }
```

Comment:
I can do it explicitly. wrap it in a try catch, and fail stage explicitly.
would that be preferable somehow?